src/S3.ts

```diff
@@ -16,43 +16,45 @@ class S3 {
   }
 
   public async initialise() {
-    const s3UrlBase: any = new S3File.Endpoint(this.url)
-    this.s3Handler = new S3File.S3({
-      endpoint: s3UrlBase,
-      accessKeyId: this.params.keyId,
-      secretAccessKey: this.params.secretAccessKey,
-      s3ForcePathStyle: s3UrlBase.protocol === 'http:',
+    const s3UrlBase: any = new URL(this.url)
```

Comment:
`s3UrlBase` seems not the best name `s3UrlObj` seems better.

Reply:
will change